docs(ai-transport): Add Troubleshooting page (#3390)
ably/docs: src/pages/docs/ai-transport/troubleshooting.mdx

---
title: "Troubleshooting"
meta_description: "Debug common Ably AI Transport problems. Channel namespace configuration, capability and token issues, history retention, turn lifecycle, cancellation, multi-device, and connection recovery."
meta_keywords: "AI Transport troubleshooting, channel namespace, capability, history retention, turn lifecycle, cancellation, reconnect"
intro: "Common AI Transport problems and how to fix them. Each entry follows the same shape: the symptom you see, what to check to confirm the cause, and the fix to apply."
---

Entries are ordered roughly by how often they cause support tickets, with the most common failure modes first.

## Channel namespace not configured for AI Transport <a id="namespace-not-configured"/>

> **Symptom:** Clients see an empty assistant message, or a message containing only the first token. The server errors on subsequent publishes. The initial `message.create` succeeds, but the `message.append` operations that AI Transport uses to stream tokens fail because the namespace does not permit appends. This is the single most common AI Transport failure.

Reviewer: I think we should include error code/error text here for searchability

**Confirm:**

- Check the server logs for an error on publish after the first token. The error references appends being rejected.
- Open the Ably dashboard for your app's settings.
- Find the channel namespace your conversations live on (for example a namespace of `conversations` should have channel names like `conversations:abc`).
- Check whether *Message annotations, updates, deletes, and appends* is enabled on that namespace.

**Fix:**

- Enable the *Message annotations, updates, deletes, and appends* rule on the namespace. See [enable message updates and deletes](/docs/ai-transport/concepts/authentication#enable-updates-deletes) for the dashboard, Control API, and CLI steps.

> - Note that enabling this rule causes messages to be persisted regardless of whether persistence is enabled, which increases usage.

Reviewer: ```suggestion
- Note that enabling this rule automatically enables [message persistence](https://ably.com/docs/storage-history/storage#all-message-persistence) for all channels in the namespace, which increases usage.
```
